### ISSUE TITLE
Powershell console opens every time I launch VS Code with a project

### ISSUE THREAD

**Issue by markdall** [microsoft/vscode#132150]
Issue Type: <b>Bug</b>

1. Update to the latest VS Code.
2. Right click VS Code icon on task bar and choose a project to load that doesn't use powershell.

Expected result: Your project loads. 
Actual result: Your project loads and the powershell console window opens, exhorting you to try the new cross-platform Powershell. 

Alternate repro steps: 
1. Navigate in the cmd.exe prompt to a project which contains no powershell code. 
2. Execute `code -n .`

Expected result: Your project loads. 
Actual result: Your project loads and the powershell console window opens, exhorting you to try the new cross-platform Powershell. 

VS Code version: Code 1.60.0 (e7d7e9a9348e6a8cc8c03f877d39cb72e5dfb1ff, 2021-09-01T10:41:52.311Z)
OS version: Windows_NT x64 10.0.21390
Restricted Mode: No

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Core(TM) i9-9900K CPU @ 3.60GHz (16 x 3600)|
|GPU Status|2d_canvas: enabled<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>oop_rasterization: enabled<br>opengl: enabled_on<br>rasterization: enabled<br>skia_renderer: enabled_on<br>video_decode: enabled<br>vulkan: disabled_off<br>webgl: enabled<br>webgl2: enabled|
|Load (avg)|undefined|
|Memory (System)|31.92GB (20.40GB free)|
|Process Argv|--file-uri file:///f%3A/wowbackups/elvui.code-workspace --crash-reporter-id 32772866-2d76-498c-8293-b5f51b9b73de|
|Screen Reader|no|
|VM|0%|
</details><details><summary>Extensions (49)</summary>

Extension|Author (truncated)|Version
---|---|---
vscode-database|baj|2.2.3
exuberant-ctags|chr|0.0.12
coddx-alpha|cod|0.3.1
bracket-pair-colorizer|Coe|1.0.61
supersharp|cra|0.1.4
vscode-markdownlint|Dav|0.43.2
githistory|don|0.6.18
python-extension-pack|don|1.6.0
xml|Dot|2.5.1
gitlens|eam|11.6.0
vscode-drawio|eig|1.0.3
git-project-manager|fel|1.7.1
vscode-pull-request-github|Git|0.30.0
language-cython|guy|0.0.6
vscode-test-explorer|hbe|2.21.1
vscode-drawio|hed|1.6.2
beautify|Hoo|1.5.0
mysql-syntax|jak|1.3.1
sharppad|jma|1.1.0
vscodeilviewer|jos|0.0.1
restructuredtext|lex|162.0.0
vscode-python-test-adapter|lit|0.7.0
MagicPython|mag|1.1.0
vscode-docker|ms-|1.16.1
csharp|ms-|1.23.15
vscode-kubernetes-tools|ms-|1.3.3
mssql|ms-|1.10.1
python|ms-|2021.9.1191016588
vscode-pylance|ms-|2021.9.0
jupyter|ms-|2021.8.2031190714
jupyter-keymap|ms-|1.0.0
remote-containers|ms-|0.194.0
remote-ssh|ms-|0.65.7
remote-ssh-edit|ms-|0.65.7
remote-wsl|ms-|0.58.2
vscode-remote-extensionpack|ms-|0.21.0
cpptools|ms-|1.6.0
powershell|ms-|2021.8.2
test-adapter-converter|ms-|0.1.2
neuron-IPE|neu|1.0.4
autodocstring|njp|0.5.4
errorlens|Phi|1.1.3
platformio-ide|pla|2.3.3
vscode-commons|red|0.0.6
vscode-yaml|red|0.22.0
cython|tcw|0.1.0
vscodeintellicode|Vis|1.2.14
vscode-todo-highlight|way|1.0.4
material-theme|zhu|3.11.4

</details><details>
<summary>A/B Experiments</summary>

```
vsliv368:30146709
vsreu685:30147344
python383cf:30185419
pythonvspyt602:30300191
vspor879:30202332
vspor708:30202333
vspor363:30204092
vstes627cf:30244335
pythonvspyt639:30300192
pythontb:30283811
pythonvspyt551:30345470
pythonptprofiler:30281270
vshan820:30294714
vstes263:30335439
pythondataviewer:30285071
pythonvsuse255:30340121
vscod805:30301674
pythonvspyt200:30340761
vscextlang:30333561
binariesv615:30325510
vsccppwtct:30329789
pythonvssor306:30344512
bridge0708:30335490
pygetstartedt2:30360495
bridge0723:30353136
vsdyn321cf:30356811
vscus224cf:30358040

```

</details>

**Comment by deqyra** [comment]
Can confirm that this is happening to me as well. This is despite having the following in my config:
```json
    "terminal.integrated.profiles.windows":{
        "Git Bash": {
            "path": "C:/Program Files/Git/git-bash.exe"
        }
    },
    "terminal.integrated.defaultProfile.windows": "Git Bash"
```
All my tasks also now run in powershell and I can't seem to have them run in any other shell. Definitely not a critical issue (in my case at least) but quite annoying. :)

**Comment by ALee-DF** [comment]
Same thing is happening to me since the latest update in #132170

**Comment by meganrogge** [comment]
Any errors in dev tools?

**Comment by ALee-DF** [comment]
I get this error on start up. Not sure if it's relevant though.
![devtool_bug](https://user-images.githubusercontent.com/31448956/131923639-46cb6148-afd4-4598-b316-62036c93edae.png)

I get no errors when switching terminals or assigning default a terminal.

**Comment by SamuelQuinones** [comment]
+1 Was going to make an issue about this, ever since the update to 1.60 no matter how I configure the settings, it will always launch powershell,

I tried nulling out the other profiles like the documentation says but to no avail:
```jsonc
{
  "terminal.integrated.profiles.windows": {
    "Git Bash": null
  }
}
```

**Comment by pinich** [comment]
Is there some kind of workaround until this issue is fixed?
I have to use git-bash as a default terminal to launch and run tasks (debugger), My environment crashes when it tries to run using Powershell which I never use.

**Comment by promethyttrium** [comment]
> Is there some kind of workaround until this issue is fixed?
> I have to use git-bash as a default terminal to launch and run tasks (debugger), My environment crashes when it tries to run using Powershell which I never use.

The workaround is the uninstall vscode and install an IDE that doesn't completely screw over its customer with a poorly tested update.... Or install the Shell Launcher extension.

Windows Update trashing people's computers seemed some much fun that the vscode team is jumping on that bandwagon too. How this is still not fixed is just insanity...

**Comment by ashelleyPurdue** [comment]
> Is there some kind of workaround until this issue is fixed?
I have to use git-bash as a default terminal to launch and run tasks (debugger), My environment crashes when it tries to run using Powershell which I never use.

Assuming I'm understanding your question correctly: you can open up a new git bash terminal even if the default is stuck on powershell.
![image](https://user-images.githubusercontent.com/8558767/132133686-89e2f99f-2bb9-4b82-af53-e18cd69dc245.png)

Then you can kill the powershell process using the trashcan icon.
![image](https://user-images.githubusercontent.com/8558767/132133721-ddc0bed0-5aec-4be4-8349-6bd09764941e.png)

It's a pain to need to do this every time you launch, but hopefully the fix will be released soon.

**Comment by pinich** [comment]
I was talking about the tasks you run using the built-in debugger which is always opened using the PowerShell who can't access the environment variables as git-bash.

![image](https://user-images.githubusercontent.com/11866550/132169155-50ba4927-9071-45cd-b507-3ccea9f71a13.png)

**Comment by bertearazvan** [comment]
I also haven't found a way around this no matter what settings I am adding. In order to fix this issue, I downgraded to `1.59` until a fix is released https://code.visualstudio.com/updates/v1_59

**Comment by weinand** [comment]
Does it help if you set the setting `terminal.integrated.automationShell.windows` to "git-bash" (or whatever shell you want to use for tasks or debug)?

**Comment by yialo** [comment]
> Does it help if you set the setting `terminal.integrated.automationShell.windows` to "git-bash" (or whatever shell you want to use for tasks or debug)?

No, it doesn't. But the following works in my case (Windows 10):

```
"terminal.integrated.automationShell.windows": "C:\\Program Files\\Git\\bin\\bash.exe"
```

**Comment by weinand** [comment]
@yialo your setting `"terminal.integrated.automationShell.windows": "C:\\Program Files\\Git\\bin\\bash.exe"`
seems to be exactly what I meant by "whatever shell you want to use for tasks or debug".

Why did you reply to my question "Does it help if you set the setting" with
> No, it doesn't.

**Comment by yialo** [comment]
> @yialo your setting `"terminal.integrated.automationShell.windows": "C:\\Program Files\\Git\\bin\\bash.exe"`
> seems to be exactly what I meant by "whatever shell you want to use for tasks or debug".
> 
> Why did you reply to my question "Does it help if you set the setting" with
> 
> > No, it doesn't.

It looks like I haven't read the question carefully enough.

**Comment by gjsjohnmurray** [comment]
If you have been affected by this issue it's likely you would have detected it before the 1.60 release if you had tried the Insiders build during endgame. In which case, maybe upvote #132369 as a way to get notified whenever an endgame begins.

**Comment by pablohs1986** [comment]
Same problem here after 1.60. Not Insider build. Could you solve it?

**Comment by markdall** [comment]
No we can't solve it, it's a defect in the program, you'll need to either wait for a release that contains the fix or to downgrade to the previous version.

**Comment by bavoceulemans** [comment]
https://github.com/microsoft/vscode/issues/123684

How does any issue first reported on may 12 and claimed to be fixed the next day only to be reported to still exist on june 18 STILL appear in a release in september?

**Comment by IngussNeilands** [comment]
Is this ticket closed ?

**Comment by albertosantini** [comment]
@IngussNeilands Yes, it is, pull merged https://github.com/microsoft/vscode/pull/132172

**Comment by mikeandtherest** [comment]
Thanks for fixing it, it is really annoying on Windows, since I'm used to gitbash. 
When can we expect to get a patch release for this it?

**Comment by Tyriar** [comment]
Today 🚀

**Comment by jonatanpedersen** [comment]
Thanks for fixing this :)

**Comment by TradeIdeasPhilip** [comment]
Not sure if this is the same bug or not.  Like the other people in this thread, I see Windows Power Shell in my interactive terminal window.  But **VS Code is _also_ using Windows Power Shell to run my build scripts**.  (shift-control-B)  And my scripts are failing; they don't like Windows Power Shell.  They should be running in git bash.

**Comment by markdall** [comment]
As far as I can tell they consider it the same bug because apparently it has the same underlying cause.

**Comment by tomshaffner** [comment]
> Today 🚀

I just checked, there's no update out and the [current release notes](https://code.visualstudio.com/updates/v1_60) still show version 1.60, so clearly the above "Today" was incorrect. Anyone _actually_ know when the release with this fix will come out?

**Comment by albertosantini** [comment]
I think the release was a bit postponed to give enough time to the devs to fix other issues.

See the candidates in the August 2021 Endgame Recovery (https://github.com/microsoft/vscode/issues/132426).

I suppose it is matter of hours, but the weekend is here.

**Comment by Tyriar** [comment]
We were planning on releasing the recovery build yesterday but the fix for another issue took a little longer to figure out so expect it Monday probably around 11-12 PST unless something goes wrong. We don't release on Fridays as half the team's in Zurich (so Saturday there) and in case something goes wrong.

**Comment by markdall** [comment]
Thanks! Also as an IT person, never release anything (or change anything or even touch anything) on a Friday, please! :)

**Comment by givensuman** [comment]
> We were planning on releasing the recovery build yesterday but the fix for another issue took a little longer to figure out so expect it Monday probably around 11-12 PST unless something goes wrong. We don't release on Fridays as half the team's in Zurich (so Saturday there) and in case something goes wrong.

Thank you for taking care of it!

**Comment by tomshaffner** [comment]
As a user, releasing something that introduces errors should also be avoided, and delaying the fix in that case means daily struggles for all of us already. This one was a real pain for me; I'll have to downgrade now.
 
 
  Am Fr., Sept. 10, 2021 at 14:15 schrieb ***@***.***>:   



We were planning on releasing the recovery build yesterday but the fix for another issue took a little longer to figure out so expect it Monday probably around 11-12 PST unless something goes wrong. We don't release on Fridays as half the team's in Zurich (so Saturday there) and in case something goes wrong.


Thank you for taking care of it!

—
You are receiving this because you commented.
Reply to this email directly, view it on GitHub, or unsubscribe.
Triage notifications on the go with GitHub Mobile for iOS or Android.

**Comment by givensuman** [comment]
> As a user, releasing something that introduces errors should also be avoided, and delaying the fix in that case means daily struggles for all of us already. This one was a real pain for me; I'll have to downgrade now. 

You can still use git bash in VSCode, it's two clicks away.

**Comment by tomshaffner** [comment]
Right, but attempting to debug python via command prompt requires me to manually enter the debug command every time until this is fixed.

  Am Fr., Sept. 10, 2021 at 15:30 schrieb ***@***.***>:   



As a user, releasing something that introduces errors should also be avoided, and delaying the fix in that case means daily struggles for all of us already. This one was a real pain for me; I'll have to downgrade now.


You can still use git bash in VSCode, it's two clicks away.

—
You are receiving this because you commented.
Reply to this email directly, view it on GitHub, or unsubscribe.
Triage notifications on the go with GitHub Mobile for iOS or Android.

**Comment by weinand** [comment]
@tomshaffner you can configure the shell used for tasks and debugging separatly via the "automationShell" setting.

**Comment by tomshaffner** [comment]
> @tomshaffner you can configure the shell used for tasks and debugging separately via the "automationShell" setting.

Does this work for running the code too?

As I understand it right now, in either case Power Shell will be opening instead of Command Prompt, which is my configured default. If you know otherwise you might want to also post it to the relevant question at https://stackoverflow.com/questions/69050842/vs-code-python-debugger-keeps-launching-powershell/69055166#69055166

**Comment by weinand** [comment]
@tomshaffner try it. If it does not work for file a bug.

**Comment by Buco11-dev** [comment]
Why do I feel like I'm being force-fed Powershell? This bug or whatever it is has left a bad aura with Powershell for me. I've now developed a phobia for Powershell. Eagerly waiting on the patch on Monday.

**Comment by robross0606** [comment]
As am I.  I left for vacation and came back to my entire JavaScript debugging setup being borked beyond use.  Not only is it opening PowerShell for debug sessions with "integratedTerminal" option, but it is now hitting breakpoints on a new tab to my (already opened) source file in read-only mode.  What a mess.

**Comment by Tyriar** [comment]
FYI conspiracy theorists, if we wanted to sneakily push powershell, introducing an annoying bug probably wouldn't be the way to do it. Ways we help powershell are by improving the product, not making it worse, eg:

- Collaborate on the extension API with the pwsh team
- Improve the pwsh experience by adding things like custom keybindings when the shell is detected as powershell
- Switched the default fallback on Windows from cmd to pwsh (this bug was caused because we couldn't find the right profile so it reverted to the fallback).

We would do the same for any shell though, I use bash primarily myself.

**Comment by Buco11-dev** [comment]
Just checking in for an update, no pressure. Has this patch been pushed yet or pending?

**Comment by Tyriar** [comment]
Release will likely happen tomorrow, got delayed to push some other fixes in. We'll update https://github.com/microsoft/vscode/issues/132426 as we're releasing.

**Comment by tomshaffner** [comment]
Again!???



-----Original Message-----

**Comment by andrewkrippner** [comment]
I just received the update and it is working properly for me now!

**Comment by marcinkubica** [comment]
I'm still getting powershell terminal open itsel when opening .ps1

**Comment by Tyriar** [comment]
> and some .ps1 files from last session

@marcinkubica do you have the powershell extension installed? If not please create a new issue with more details as this one is definitely fixed.

**Comment by marcinkubica** [comment]
> > and some .ps1 files from last session
> 
> @marcinkubica do you have the powershell extension installed? If not please create a new issue with more details as this one is definitely fixed.

Hi, I do have PowerShell extension installed.

**Comment by Tyriar** [comment]
@marcinkubica oh ok, well it opens its own terminal when you open a ps1 file

**Comment by ramiro-e** [comment]
I just got the 1.60.1 update and it has been fixed

**Comment by WORMSS** [comment]
Confirmed 1.60.1 fixed this for me. (Though, I had to re-set my default)

**Comment by mahendra2448** [comment]
I just add new profile in settings.json like this and it's works
`
    "terminal.integrated.profiles.windows": {
        "Cmder": {
            "path": "C:\\Windows\\System32\\cmd.exe",
            "args": [
                "/K",
                "C:\\cmder\\vendor\\bin\\vscode_init.cmd"
            ]
        }
    },
    "terminal.integrated.defaultProfile.windows": "Cmder",
    "terminal.external.windowsExec": "C:\\cmder\\Cmder.exe",
`

**Comment by mahendra2448** [comment]
just add this to the `settings.json` , hope this helps someone
```
{
    "terminal.integrated.profiles.windows": {
        "Cmder": {
            "path": "C:\\Windows\\System32\\cmd.exe",
            "args": [
                "/K",
                "C:\\cmder\\vendor\\bin\\vscode_init.cmd"
            ]
        }
    },
    "terminal.integrated.defaultProfile.windows": "Cmder",
    "terminal.external.windowsExec": "C:\\cmder\\Cmder.exe",
}
```

**Comment by pebeid** [comment]
Hello
This is still an issue for me even with version 1.16.2 

![image](https://user-images.githubusercontent.com/48051935/135192711-3cb80afe-ae08-44aa-8342-d2d000b7e215.png)
![image](https://user-images.githubusercontent.com/48051935/135192761-1eab8d6e-1d14-4e4b-b41e-1fc0d5fa6a5a.png)
